src/textual/app.py

```diff
@@ -4314,3 +4314,15 @@ def _on_delivery_failed(self, event: events.DeliveryComplete) -> None:
             self.notify(
                 "Failed to save screenshot", title="Screenshot", severity="error"
             )
+
+    def get_keymap(self) -> Keymap:
```

Comment:
Given the keymap is a wrapper around a dict, could be just use a dict?

It also looks like it would not be a problem to modify the dict on the fly. Perhaps we should just expose a dict via an attribute. And the user could replace / update `self.keymap` as needed?

Reply:
I wasn't really intending this to be mutable at this stage to be honest.

I wrapped it to keep the mapping immutable and also leave flexibility for extension in the future without a breaking change. Any utility methods/helper properties could be encapsulated inside it, etc.

I'm worried if we make it mutable then we run into some of the same problems we get with mutable reactives. If I update a key, in the keymap, would I need to refresh the footer and keypanel? I've no major problem with it being mutable, but would probably prefer if we had some means of doing it that kept the footer/help panel up-to-date when that mutation happens.

What about an reactive attribute `App.keymap` of type `Keymap`, which remains as-is. The `App.watch_keymap` watcher would refresh any key panels and footers as required?

Reply:
How about making the attribute private. Then having `update_keymap` and `replace_keymap` methods? That would avoid the issues with mutability.

Reply:
Are you opposed to having a `Keymap` object? If so I would like to understand why 😄 

Using primitive dicts makes me sadder than when I see a walrus operator inside a complex expression :( - it's harder to document, means any utility methods/properties we want would need to be crammed in `App`, we lose typing benefits if we want to pass it around, it's less readable because we if I see `dict[str, str]` then I don't know what the strings are (and if I alias them to make it clearer then I need to import both aliases everywhere I use the dict), if we wanted to extend the functionality we need to do a breaking change, we can't add custom reprs etc.

Reply:
Yeah, apposed I'm afraid.

If its private, then the `Keymap` would just exist to wrap a dict and make it immutable. I don't think we've wrapped an atomic type anywhere else with the only goal of making it immutable. The closest I can think of is NodeList, but that adds additional functionality on top of a list.

We can still type methods are receiving a TypeAlias. I don't think it is a major hassle to have to import a type. We do that a lot already.

Ultimately, while it is just a dict, I think it should remain as such.

I can't see how it would result in a breaking change if we needed to go from a dict, to a dict-like object in the future.

Can chat about it if you like. Just in case we aren't on the same page.

Reply:
Let's agree to disagree. I feel like a gave a pretty strong list of reasons why I think an object is better and you didn't really say concretely why you think a dict is better... but it's your call, I'll make the change tomorrow 😄